Makefile

```diff
@@ -2,13 +2,19 @@ VERSION = 0.27.1
 VERBOSE_FLAG = $(if $(VERBOSE),-verbose)
 CURRENT_REVISION = $(shell git rev-parse --short HEAD)
 
+GOOS   ?= $(shell go env GOOS)
+GOARCH ?= $(shell go env GOARCH)
+BINDIR  = build/$(GOOS)/$(GOARCH)
+
 all: lint cover testconvention rpm deb
 
+$(BINDIR)/mackerel-plugin-%: mackerel-plugin-%/lib/*.go
+	@if [ ! -d $(BINDIR) ]; then mkdir -p $(BINDIR); fi
+	go build -ldflags="-s -w" -o $@ ./`basename $@`
+
 build: deps
-	mkdir -p build
 	for i in mackerel-plugin-*; do \
-		go build  -ldflags="-s -w" -o build/$$i \
-		`pwd | sed -e "s|${GOPATH}/src/||"`/$$i; \
+	  make $(BINDIR)/$$i; \
 	done
 
 build/mackerel-plugin: deps
```

Comment:
Should we also change directory structure of `build/mackerel-plugin` to use `BINDIR` ?